pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -414,17 +382,22 @@ func (c *Controller) onNodeEvent(obj interface{}, event model.Event) error {
 	return nil
 }
 
-func nodeEquals(a, b kubernetesNode) bool {
-	return a.address == b.address && a.labels.Equals(b.labels)
-}
-
-func isNodePortGatewayService(svc *v1.Service) bool {
-	_, ok := svc.Annotations[kube.NodeSelectorAnnotation]
-	return ok && svc.Spec.Type == v1.ServiceTypeNodePort
-}
+// Filter func for filtering out objects during update callback
+type FilterOutFunc func(old, cur interface{}) bool
 
 func registerHandlers(informer cache.SharedIndexInformer, q queue.Instance, otype string,
-	equalsFunc func(old, cur interface{}) bool, handler func(interface{}, model.Event) error) {
+	handler func(interface{}, model.Event) error, filter FilterOutFunc) {
+	if filter == nil {
+		filter = func(old, cur interface{}) bool {
```

Comment:
what is the advantage of defining this filter func inside of the `registerHandlers` func rather than defining this outside and pass it to this function? That way we do not have to handle nil and from call site it is clear that we are doing a version based equality check.

Any benefits of defining here?

Reply:
It is just to dedup passing the same filter function everywhere

Reply:
Ok. I was thinking of defining a function like this and passing `VersionFilterFunc` instead of nil and removing nil check here. - But NBD
 func VersionFilterFunc(old, cur interface{}) bool {
oldObj := old.(metav1.Object)
			newObj := cur.(metav1.Object)
			// TODO: this is only for test, add resource version for test
			if oldObj.GetResourceVersion() == "" || newObj.GetResourceVersion() == "" {
				return false
			}
			return oldObj.GetResourceVersion() == newObj.GetResourceVersion()
}

Reply:
I prefer what Rama suggest as well, but not a big concern

Reply:
SGTM. I can do that once we need other filters